# $\color{violet}{\text{Welcome To Data Science Camp!}}$

## $\color{olive}{\text{What is Data Science?}}$

- Data science is a  field that uses techniques from statistics, computer science, and domain expertise to extract insights and knowledge from data
- Data science includes:
  - Data Collection
  - Data Cleaning
  - Exploratory Data Analysis (EDA)
    - Visualizing and summarizing data to understand patterns.
  - Statistical Modeling & Machine Learning
- In this camp we will focus on Exploratory Data Analysis!


# $\color{cyan}{\text{Learning to Work With R}}$

- We will be analyzing the datasets using a computer programming language called R.

- Before we begin our data analysis, lets learn some of the basics to the coding language.

- We can perform basic operation in R such as addition, multiplication, exponents, subtraction and division.

- **NOTE**: To put comments into R we can use # then type whatever your comment is. This is a line in the code that is not run.

- **NOTE**: To run a chunk of code in this notebook, hit the play button to the left of the code chunk.

- Let's try it! If we want to find the answer to $(5+2)^2 -6 \div 3$ we can input this into R as:

In [ ]:
#this is a comment! This will not run. Rather, it is a note for the coder.
(5+2)^2 -6/3

[1] 47

### $\color{magenta}{\text{Your Turn!}}$

 In the code chunk below, compute $(5-2)^2 +12 \div 4$

In [ ]:
#enter your code here.

## $\color{olive}{\text{Functions}}$




Great work! Now we will talk about funcitons.

A function in coding is a chunk of code that performs a specific task. For example, in R the function sqrt() will find the square root of a number.

The value we put into the brackets is the "argument" of the function. Let's find the square root of 9, for example.

In [ ]:
sqrt(9)

### $\color{magenta}{\text{Your Turn!}}$

 Find the square root of 49 using the sqaure root function.

In [ ]:
#enter your code here

Great job! In R we can also print out words using the print function, and putting the string of words in quotations. For example we can make R return "Hello World!"

In [ ]:
print("Hello World") #Run this line of code by hitting the "play" button

### $\color{magenta}{\text{Your Turn!}}$

 Make R print out "My name is [your name here]"

In [ ]:
#Enter your code here

## $\color{olive}{\text{ Mean, Median, Mode, and Range}}$

There are many things a data scientist might be interested in. Some summary values that are important include: Mean, Median, Mode, and Range.

To investigate these concepts we will use some fake data. Our data consists of 10 values:
  - 19, 17, 12, 17, 20, 18, 3, 8, 4, 1

We will store this data in a **vector** that we will assign a name. This is called a **variable**
  - A variable is like a name you give to something so the computer can remember it for you.
  - In R, you use a variable when you want to save a number, word, or idea and use it later.
  - A vector is like a list of things that all go together. It’s a way to keep a bunch of items together.

In [8]:
#Let's make a vector, but not save it as a variable
c(19, 17, 12, 17, 20, 18, 3, 8, 4, 1)
#We want to be able to access this information later, so we will save it as a variable called "x"
x = c(19, 17, 12, 17, 20, 18, 3, 8, 4, 1)

[1] 19 17 12 17 20 18  3  8  4  1

### $\color{cyan}{\text{Mode}}$

- Mode means the most common value in a group of things. It’s the one that shows up the most!
- From my fake data "x", we can see the mode is 17. It shows up twice, while all the other data shows up once.
- We can get the mode in R by using the table() function.
  - The table() function counts how many times each item shows up in a list.
  - It makes a mini chart that tells you:
    - What items are in the list
    - How often each one appear

In [9]:
table(x)

x
 1  3  4  8 12 17 18 19 20 
 1  1  1  1  1  2  1  1  1 

### $\color{cyan}{\text{Mean}}$
- The mean is what most people call the average. It tells you what the numbers are like on average.
-  How Do You Find the Mean?
  - Add up all the numbers.
  - Divide by how many numbers there are.

For our example, the average or mean is: $(19+ 17+ 12+ 17+ 20+ 18+ 3+ 8+ 4+ 1) \div 10 = 11.9$

- In R we can use the mean() function to easily calculate the mean of a vector.




In [3]:
mean(x)

[1] 11.9

### $\color{cyan}{\text{Median}}$

- The median is the middle number in a list of numbers.
- First, you put the numbers in order (from smallest to biggest). Then, you find the one in the middle. That’s the median!
- If there are two numbers in the middle, you take the average of them. That is the median!

For our example, we sort or data smallest to largest:
  - 1, 3, 4, 8, 12, 17, 17, 18, 19, 20
Here 12 and 17 are both in the middle, so the median is $(12+17)\div 2 = 14.5$

We can use the R function median() to easily compute the median of our vector.

In [4]:
median(x)

[1] 14.5

### $\color{cyan}{\text{Range}}$

The range tells you how spread out the numbers are. It is the difference between the biggest and smallest number.

For our dataset the range is $20-1=19$

We can use the function max() and min() in R to find out maximumn and minimum values, then we subtract them.

In [6]:
max(x)-min(x)

[1] 19

### $\color{magenta}{\text{Your Turn!}}$



Consider a collection of numbers:  2, 3, 10, 4, 19, 8, 4, 12, 4, 19,  5, 1

Use R to find the mean, median, mode and range of the data!

In [ ]:
#Enter your code here.

## $\color{olive}{\text{Collecting Data}}$

I am interested in some information about my friends! I will ask my friends the following information in a survery:
  - Name, Age, number of pets, average daily screen time, number of siblings, height, how many people you follow on your favourite social media, favourite social media

After handing out a survey we gathered the following information:

- Danika
  - Age: 27
  - Number of pets: 1
  - Average daily screen time: 9.5 hours
  - Number of siblings: 1
  - height: 172cm
  - How many people you follow on your favourite social media: 452
  - Favourite social media: instagram
- Thuntida
  - Age: 40
  - Number of pets: 1
  - Average daily screen time: 8hours
  - Number of siblings: 1
  - height: 158cm
  - How many people you follow on your favourite social media: 310
  - Favourite social media: instagram
- Placida
  - Age: 40
  - Number of pets: 0
  - Average daily screen time: 4
  - Number of siblings: 2
  - height: 162.5cm
  - How many people you follow on your favourite social media: 112
  - Favourite social media: facebook
- Marco
  - Age: 47
  - Number of pets: 3
  - Average daily screen time: 7.2hours
  - Number of siblings: 2
  - height: 181cm
  - How many people you follow on your favourite social media: 300
  - Favourite social media: X
- Alvin
  - Age: 32
  - Number of pets: 1
  - Average daily screen time: 15.3hours
  - Number of siblings: 2
  - height: 155cm
  - How many people you follow on your favourite social media: 232
  - Favourite social media: TikTok
- Veronica
  - Age: 31
  - Number of pets: 2
  - Average daily screen time: 5.2hours
  - Number of siblings: 3
  - height: 170cm
  - How many people you follow on your favourite social media: 83
  - Favourite social media: TikTok
- Don
  - Age: 41
  - Number of pets: 0
  - Average daily screen time: 11.1 hours
  - Number of siblings: 1
  - height: 151cm
  - How many people you follow on your favourite social media: 230
  - Favourite social media: instagram
- Ryan
  - Age: 35
  - Number of pets: 1
  - Average daily screen time: 13.3 hours
  - Number of siblings: 3
  - height: 196cm
  - How many people you follow on your favourite social media: 336
  - Favourite social media: TikTok
- Maggie
  - Age: 42
  - Number of pets: 2
  - Average daily screen time: 10.5 hours
  - Number of siblings: 0
  - height: 147cm
  - How many people you follow on your favourite social media: 269
  - Favourite social media: facebook
- Aaron
  - Age: 36
  - Number of pets: 4
  - Average daily screen time: 10.8 hours
  - Number of siblings: 2
  - height: 156cm
  - How many people you follow on your favourite social media: 244
  - Favourite social media:snapchat
- Scott
  - Age: 29
  - Number of pets: 2
  - Average daily screen time: 7.3 hours
  - Number of siblings: 5
  - height: 141cm
  - How many people you follow on your favourite social media: 176
  - Favourite social media: snapchat

  **We need to put this data into R! We will start by storing the information in vector variables. I will name these variables something informative.**

  **We need to make sure we enter the information in the correct order**

In [2]:
Name = c("Danika", "Thuntida", "Placida", "Marco", "Alvin", "Veronica", "Don", "Ryan", "Maggie", "Aaron", "Scott")
Age = c(27, 40, 40, 47, 32, 31, 41, 35, 42, 36, 29)
Pets = c(1, 1, 0, 3, 1, 2, 0, 1, 2, 4, 2)
ScreenTime = c(9.5, 8, 4, 7.2, 15.3, 5.2, 11.1, 13.3, 10.5, 10.8, 7.3)
Siblings = c(1, 1, 2, 2, 2, 3, 1, 3, 0, 2, 5)
Height = c(172, 158, 162.5, 181, 155, 170, 151, 196, 147, 156, 141)
Following = c(425, 310, 112, 300, 232, 83, 230, 336, 269, 244, 176)
SocialMedia = c("IG", "IG", "FB", "X", "TikTok", "TikTok", "IG", "TikTok", "FB", "SC", "SC")

Now that we have the data all stored in vectors, we can combine the vectors into what we call a data frame. Each row will contain information about about a different sample/person. Each column represents a type of information.

In [3]:
#Put the information into a dataframe called dataframe1
dataframe1= data.frame(Name,Age,Pets,ScreenTime,Siblings,Height,Following,SocialMedia)
#Now we cant print out the dataframe to look at it

print(dataframe1)

       Name Age Pets ScreenTime Siblings Height Following SocialMedia
1    Danika  27    1        9.5        1  172.0       425          IG
2  Thuntida  40    1        8.0        1  158.0       310          IG
3   Placida  40    0        4.0        2  162.5       112          FB
4     Marco  47    3        7.2        2  181.0       300           X
5     Alvin  32    1       15.3        2  155.0       232      TikTok
6  Veronica  31    2        5.2        3  170.0        83      TikTok
7       Don  41    0       11.1        1  151.0       230          IG
8      Ryan  35    1       13.3        3  196.0       336      TikTok
9    Maggie  42    2       10.5        0  147.0       269          FB
10    Aaron  36    4       10.8        2  156.0       244          SC
11    Scott  29    2        7.3        5  141.0       176          SC


### $\color{magenta}{\text{Your Turn!}}$

Collect data from all of the campers, and put this information into a dataframe called dataframe2.

In [ ]:
#Enter your code here

## $\color{olive}{\text{Extracting Data}}$

- If we want to look at a particular row of data we can run the line of code:
  - dataframe[row,]

For example if we want to look at the second row of data:

In [4]:
dataframe1[2,]

Name     Age Pets ScreenTime Siblings Height Following SocialMedia
2 Thuntida 40  1    8          1        158    310       IG

- If we want to extract a particular column from our dataframe we have two ways to do this:
  - dataframe[,column]
  - dataframe$name of column

Lets extract the 4th column, which is also called ScreenTime

In [5]:
dataframe1[,4]
dataframe1$ScreenTime

[1]  9.5  8.0  4.0  7.2 15.3  5.2 11.1 13.3 10.5 10.8  7.3

[1]  9.5  8.0  4.0  7.2 15.3  5.2 11.1 13.3 10.5 10.8  7.3

 - If we want to get the value from the 4th row, and 3rd column we can do that in R as follows:

In [6]:
dataframe1[4,3]

[1] 3

We can apply some functions to our data as well! For example I can extract, the mean, median, mode and range of the Age variable.

In [10]:
mean(dataframe1$Age)
median(dataframe1$Age)
table(dataframe1$Age)
max(dataframe1$Age) - min(dataframe1$Age)


[1] 36.36364

[1] 36


27 29 31 32 35 36 40 41 42 47 
 1  1  1  1  1  1  2  1  1  1 

[1] 20

The mean is 36.36, the median is 36, the mode is 40, and the range of the data is 20 years.

### $\color{cyan}{\text{Maximums and Minimums}}$

- I can find the maximum in a dataset using max()
- I can find the minimum in a dataset using min()
- If I want to know where the maximum is I can use which.max()
- If I want to know where the minimum is I can use which.min()


For example, let's find the tallest person, the shortest person, and their heights!

In [19]:
#find the biggest height
max(dataframe1$Height)
#the tallest person is 196cm
#now I want to know which entry this is
which.max(dataframe1$Height)
#I can now grab the name of this person
dataframe1$Name[which.max(dataframe1$Height)]
#The tallest person is Ryan!

#find the smallest height
min(dataframe1$Height)
#the shortest person is 141cm
#now I want to know which entry this is
which.min(dataframe1$Height)
#I can now grab the name of this person
dataframe1$Name[which.min(dataframe1$Height)]
#The tallest person is Scott!

[1] 196

[1] 8

[1] "Ryan"

[1] 141

[1] 11

[1] "Scott"

### $\color{magenta}{\text{Your Turn!}}$

Using the data you collected and stored in dataframe2, find and report the:

- mean, median, mode, and range of age
- Who follows the least people on social media? How many people do they follow?
- Who follows the most people on social media? How many people do they follow?

In [ ]:
#Enter your code here

Double Click here to type out your answer!

# $\color{violet}{\text{Module 2: Data Visualization }}$

- histograms
- standard deviation
  - activity for standard deviation - skittles
  - skews, mean, median, mode on the histogram
- scatter plots Bivariate -introuduce new dataset

# $\color{violet}{\text{Module 3: Analyzing data with Linear Regression}}$

- correlation, SLR,
